### ISSUE TITLE
[rollup-plugin-html] extractAssets from custom tagName

### ISSUE THREAD

**Issue by gdbaldw** [modernweb-dev/web#1558]
Currently, `extractAssets = boolean`. But, sometimes we'd like to extract more assets, as has been identified at https://github.com/modernweb-dev/rocket/issues/162 and at https://github.com/modernweb-dev/web/pull/1008

So, I wrote a pull request https://github.com/modernweb-dev/web/pull/1541 such that `extractAssets = boolean | tagAndAttribute[]`

## Example Usage

```js
const html = require('../../dist/index').default;

module.exports = {
  input: 'demo/spa/index.html',
  output: {
    dir: './demo/dist',
  },
  plugins: [html({
    absoluteBaseUrl: 'http://localhost:8000',
    extractAssets: [
      {tagName: 'my-first-el', attribute: 'my-src'},
      {tagName: 'my-second-el', attribute: 'my-src'},
    ]
  })],
};
```

**Comment by bennypowers** [comment]
adjacent: prefetching assets for non-standard script types, and finding assets in the fragment of `<template>` tags

**Comment by gdbaldw** [comment]
> adjacent: prefetching assets for non-standard script types, and finding assets in the fragment of `<template>` tags

What would extractAssets interface look like? Add to the current pull request? (it's already kind of bloated, no?) I would appreciate guidance in moving this forward.

**Comment by bennypowers** [comment]
I don't think you have to change much here. If you really want, add a test case for assets in template fragments

```js
it('extracts assets in template fragments', function() {
  const fixture = `
    <img src="a.png">
    <template>
      <img src="b.png">
      <template>
        <img src="c.png">
      </template>
    </template>
  `;
  // ...
  expect(assets.length).to.equal(3);
}); 
```

**Comment by gdbaldw** [comment]
> I don't think you have to change much here. If you really want, add a test case for assets in template fragments
> 
> ```js
> it('extracts assets in template fragments', function() {
>   const fixture = `
>     <img src="a.png">
>     <template>
>       <img src="b.png">
>       <template>
>         <img src="c.png">
>       </template>
>     </template>
>   `;
>   // ...
>   expect(assets.length).to.equal(3);
> }); 
> ```

Correct! Just one recursive function, done. Ready for code review?
